lib/matplotlib/tests/test_backend_gtk3.py

```diff
@@ -0,0 +1,58 @@
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+
+from matplotlib.externals import six
+from matplotlib.externals.six import unichr
+from matplotlib import pyplot as plt
+from matplotlib.testing.decorators import cleanup, switch_backend
+from matplotlib.testing.decorators import knownfailureif
+from matplotlib._pylab_helpers import Gcf
+import copy
+
+try:
+    # mock in python 3.3+
+    from unittest import mock
+except ImportError:
+    import mock
+
+try:
+    from matplotlib.backends.gtk3_compat import Gtk, Gdk, GObject, GLib
+    HAS_GTK3 = True
+except ImportError:
+    HAS_GTK3 = False
+
+
+def simulate_key_press(canvas, key, modifiers=[]):
+    event = mock.Mock()
+
+    keyval = [k for k, v in six.iteritems(canvas.keyvald) if v == key]
+    if keyval:
+        keyval = keyval[0]
+    else:
+        keyval = ord(key)
+    event.keyval = keyval
+
+    event.state = 0
+    for key_mask, prefix in canvas.modifier_keys:
+        if prefix in modifiers:
+            event.state |= key_mask
+
+    canvas.key_press_event(None, event)
+
+
+@cleanup
+#@knownfailureif(not HAS_GTK3)
```

Comment:
Yes, just for Travis testing, it makes it easier to spot import mistakes.  With this line commented out, if everything goes well all but python 2.6 should pass.  With this line in, it can mask import errors on builds that should pass.

I guess I could replace this with a knownfailiureif(sys.version_info < (2, 7)), which would ensure Travis always flags errors, but for people testing on their own machines without GTK3 installed it wouldn't work so well...  Do you have an opinion on this?

Of course the middle way, perhaps we should add (or perhaps it already exists, but I haven't seen), an ``isTravis`` variable so we can a test can only fail on developers machines but not on Travis...